addons/website/static/src/snippets/s_website_form/options.js

```diff
@@ -1214,6 +1220,16 @@ options.registry.WebsiteFieldEditor = FieldEditor.extend({
                 <we-button data-select-data-attribute="!fileSet">${_t("Is not set")}</we-button>
             </we-select>
         `)[0]);
+        const recursiveFindCircular = (el) => {
+            if (el.closest('.s_website_form_field').dataset.visibilityDependency === this._getFieldName()) {
+                return true;
+            }
+            const dependencyEl = this._getDependencyEl(el);
```

Comment:
@rdeodoo I push-forced with a typo fix in the commit message (it's -> its) + a rebase, then I actually went and tested the branch and it in fact does not work, not sure why. I tried to create a circular dependency between 3 fields... the system properly prevented me to do so... but then I continue my chain with 4 fields and I was able to do it.

After a small check, I am not sure everything is right here:

1. The first time `recursiveFindCircular` is called is with a `.s_website_form_field` element
2. The first line of the function does `.closest('.s_website_form_field')` (which is ok but useless in that case)
3. `_getDependencyEl` is called and read `.dataset.visibilityDependency` on that `.s_website_form_field` element... but return a `.s_website_form_input` element (which is normally a descendant of a `.s_website_form_field`)
4. We call `recursiveFindCircular` with that `.s_website_form_input` element instead of `.s_website_form_field` element.
5. Like (2), the first line of the function does `.closest('.s_website_form_field')`... which is now actually needed
6. Like (3), we call `_getDependency` but it reads `.dataset.visibilityDependency` on the `.s_website_form_input` element which is probably wrong...

What I wonder is how it worked up until I tried 4 fields circular dependency ^^ Maybe the fix is about moving the `.closest('.s_website_form_field')` on the first line of `recursiveFindCircular` to its last line (when it called recursively), I let you check :+1:

Reply:
Whoops, indeed. Fixed with [this](https://github.com/odoo/odoo/compare/8f2771863398afab7e9eca44c32e68f146b49a53..b8cbc134f7621922e477a71e06e71412360e7acb), it was due to last push force [adaptation](https://github.com/odoo/odoo/compare/d092dd791d0eb0157500203cb0bb9f20a1600732..86684463cd119960223101d3cf714165a143cfb2#diff-8b11d359e1213470895c7a9d28b45424520045089330be5713e9f3865ad16406R1224) where I merged the direct and indirect check in the recursive method, damn.

I also tried max 3 fields so didn't see the error:
- no field left (1>3 2>3 3>1)
- field left but would create circular (1>2>3>1)

I just triple checked with those multiple 3 fields cases, 4 fields, 5 fields, 1 field